options.js

```diff
@@ -1,42 +1,65 @@
-chrome.storage.local.get(
-    ["options", "exclusionArray"],
-    ({
-        options = {
-            moveTabs: true,
-            effectWindows: false,
-            effectTabGroups: false,
-        },
-        exclusionArray = [],
-    }) => {
-        const { moveTabs, effectWindows, effectTabGroups } = options;
+
+/**
+ * Read from local storage in async instead of using callbacks
+ * Pass in a single string key or an array of keys to retrieve multiple values
+ * 
+ * @param {string|array<string>} key 
+ * @returns {object}
+ */
+const getLocalStorageKey = key => {
+    let storageKey = key;
+    if (typeof storageKey !== 'array') storageKey = [storageKey];
+
+    return new Promise((resolve, reject) => {
+        try {
+            chrome.storage.local.get(key, resolve)
+        } catch (err) {
+            reject(err);
+        }
+    });
+}
+
+/**
+ * Set an object in local storage to the given value
+ * Local Stoarge objects key is based on the Object key passed in
+ * @param {Object} val 
+ */
+const setLocalStorageValue = val => chrome.storage.local.set(val);
+
+/**
+ * Upsert into options local storage namespace
+ * Local Stoarge objects key is based on the Object key passed in
+ * @param {Object} val 
+ */
+const updateOptions = async (updatedValues) => {
+    const { options } = await getLocalStorageKey();
+    const updatedOptions = {
+        ...options,
+        ...updatedValues,
+    }
+    setLocalStorageValue({ options: updatedOptions });
+}
```

Comment:
So naturally there are many ways to accomplish this general upsert idea. You can even mess with where we use our shorthand. For example:
```suggestion
const updateOptions = async (updatedValues) => {
    const { options: oldOptions } = await getLocalStorageKey();
    const options = {
        ...oldOptions ,
        ...updatedValues,
    }
    setLocalStorageValue({ options });
}
```

Maybe this second way does look prettier.